model/riscv_insts_zbkb.sail

```diff
@@ -0,0 +1,104 @@
+/* ****************************************************************** */
+union clause ast = ZBKB_RTYPE : (regidx, regidx, regidx, brop_zbkb)
+
+mapping clause encdec = ZBKB_RTYPE(rs2, rs1, rd, RISCV_PACK) if haveZbkb()
+  <-> 0b0000100 @ rs2 @ rs1 @ 0b100 @ rd @ 0b0110011 if haveZbkb()
+
+mapping clause encdec = ZBKB_RTYPE(rs2, rs1, rd, RISCV_PACKH) if haveZbkb()
+  <-> 0b0000100 @ rs2 @ rs1 @ 0b111 @ rd @ 0b0110011 if haveZbkb()
+
+mapping zbkb_rtype_mnemonic : brop_zbkb <-> string = {
+  RISCV_PACK  <-> "pack",
+  RISCV_PACKH <-> "packh"
+}
+
+mapping clause assembly = ZBKB_RTYPE(rs2, rs1, rd, op)
+  <-> zbkb_rtype_mnemonic(op) ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+function clause execute (ZBKB_RTYPE(rs2, rs1, rd, op)) = {
+  let rs1_val = X(rs1);
+  let rs2_val = X(rs2);
+  let result : xlenbits = match op {
+    RISCV_PACK  => if sizeof(xlen) == 32
+                   then rs2_val[15..0] @ rs1_val[15..0]
+                   else rs2_val[31..0] @ rs1_val[31..0],
```

Comment:
Just use `sizeof(xlen)/2 - 1` for the upper bound, no if needed

Reply:
Okay... I'll change this.

Reply:
I'm allergic to using /2 vs  >>1; should I not care, or is this an integer vs. bitvector thing and we should be using /2?

Reply:
`>>` isn't defined for integers currently in model/prelude.sail, only for bitvectors, though there's no reason you couldn't do so I guess. But really `/` should be used for things that are logically a division, and `>>` for things that are logically a bit shift (ignoring the fact that the two are equivalent in certain cases), with the compiler left to optimise division to a bit shift as necessary. In this case it'll all be constant-folded anyway as it's a constant divided by a constant.

Also, /2 and >>1 are not the same thing on negative values, they round in different directions.

Reply:
(i.e. embrace your inner mathematician rather than your inner hardware engineer, we're not in the grotty world of verilog, we're in a nice high-level language that's trying to specify things in as clear and pure a way as we can)

Reply:
Same division issue will be there while using `sizeof(xlen)/2 - 1` as upper bound